awscli/customizations/s3/s3handler.py

```diff
@@ -385,151 +396,112 @@ def __init__(self, session, params, result_queue=None,
 
-    def _enqueue_multipart_tasks(self, filename, payload=None):
-        num_uploads = 1
-        if filename.operation_name == 'upload':
-            num_uploads = self._enqueue_multipart_upload_tasks(filename,
-                                                               payload=payload)
-        elif filename.operation_name == 'download':
-            num_uploads = self._enqueue_range_download_tasks(filename)
-        return num_uploads
+        future = manager.download(
+            fileobj=StdoutBytesWriter(), bucket=bucket,
+            key=key, extra_args=params)
 
-    def _enqueue_range_download_tasks(self, filename, remove_remote_file=False):
+        return self._process_transfer(future)
 
-        # Create the context for the multipart download.
-        num_downloads = int(filename.size / self.chunksize)
-        context = tasks.MultipartDownloadContext(num_downloads)
+    def _upload(self, manager, bucket, key):
+        """
+        Upload stdin using to the specified location.
 
-        # No file is needed for downloading a stream.  So just announce
-        # that it has been made since it is required for the context to
-        # begin downloading.
-        context.announce_file_created()
+        :type manager: s3transfer.manager.TransferManager
+        :param manager: The transfer manager to use for the upload.
 
-        # Submit download part tasks to the executor.
-        self._do_enqueue_range_download_tasks(
-            filename=filename, chunksize=self.chunksize,
-            num_downloads=num_downloads, context=context,
-            remove_remote_file=remove_remote_file
-        )
-        return num_downloads
+        :type bucket: str
+        :param bucket: The bucket to upload the stream to.
 
-    def _enqueue_multipart_upload_tasks(self, filename, payload=None):
-        # First we need to create a CreateMultipartUpload task,
-        # then create UploadTask objects for each of the parts.
-        # And finally enqueue a CompleteMultipartUploadTask.
+        :type key: str
+        :param key: The name of the key to upload the stream to.
+
+        :return: A CommandResult representing the upload status.
+        """
+        # Upload chunksizes are restricted, so we need to do some massaging
+        # to get them right.
         if self.params['expected_size']:
             # If we have the expected size, we can calculate an appropriate
             # chunksize based on max parts and chunksize limits
-            chunksize = find_chunksize(int(self.params['expected_size']),
-                                       self.chunksize)
+            chunksize = find_chunksize(
+                int(self.params['expected_size']),
+                self.config.multipart_chunksize)
         else:
             # Otherwise, we can still adjust for chunksize limits
-            chunksize = adjust_chunksize_to_upload_limits(self.chunksize)
+            chunksize = adjust_chunksize_to_upload_limits(
```

Comment:
Hmm we need to fix this in ``s3transfer``because we do not readjust the chunksize. It might be good to put a TODO to make sure we remove this once we we port it to s3transfer.